ai/drone_configuration.py

```diff
@@ -198,14 +220,17 @@ async def toggle_parameter(context,
     channel_webhook = await webhook.get_webhook_for_channel(context.channel)
 
     for drone in drones:
-        trusted_users = get_trusted_users(drone.id)
-        if has_role(context.author, HIVE_MXTRESS) or context.author.id in trusted_users:
+        if can_toggle_permissions_for(context.author, drone):
             message = ""
             if is_toggle_activated(drone):
                 update_droneOS_parameter(drone, toggle_column, False)
                 await drone.remove_roles(role)
                 message = toggle_off_message()
 
+                still_configured = is_configured(drone)
+                if not still_configured:
+                    update_droneOS_parameter(drone, "can_self_configure", True)
```

Comment:
This will unfortunately not work, as the table `drone` currently does not have a column `can_self_configure`.

This is a decent approach though, so this could be resolved by adding the column to the current schema. Please refer to https://github.com/HexCorpProgramming/HexCorpDiscordAI/wiki/Database#migration-files to learn about how we change the DB schema. It mostly boils down to adding a new SQL file to the directory `res/db` with an ALTER TABLE statement.

Of course you can ask any questions you might have in the dev chat~

Reply:
Resolved. (it _____s)

Reply:
Yes, exactly like that. This should work now.